sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -55,6 +56,26 @@ def test_mean_shift():
     assert_equal(n_clusters_, n_clusters)
 
 
+def test_mean_shift_negative_bandwidth():
+    bandwidth = -1
+    ms = MeanShift(bandwidth=bandwidth)
+    msg = ("bandwidth needs to be greater than zero or None,"
+           "            got -1.000000")
+    assert_raise_message(ValueError, msg, ms.fit, X)
+
+
+@pytest.mark.parametrize("bandwidth, cluster_all, expected", [
+    (1.2, True, 3),
+    (1.2, False, 4)
+])
+def test_eval(bandwidth, cluster_all, expected):
```

Comment:
what do you mean by calling this "eval"? Can't we just paramertrize `test_mean_shift` above, rather than adding a new test?

Reply:
But ideally we should also test that `cluster_all=False` is actually effective at allowing some points to be left unclustered. Create a dataset where a point will be left with label -1 to test this properly.

Reply:
@jnothman  fixed as suggested